### ISSUE TITLE
sass-binary-dir does not use node-sass version

### ISSUE THREAD

**Issue by Johanneke** [sass/node-sass#2583]
For systems without free internet access, we need to provide alternative ways to find the "sass-binary". 
When specifying a proxy site using `sass-binary-site`, it will append the url with a path segment "v + node-sass version" (eg `v4.9.2`) and then a file with a specific name, eg `win32-ia32-64_binary.node`.
When specifying a (local) directory using `sass-binary-dir`, it will look for a directory named `win32-ia32-64`, and then simply a file named `binding.node`. The version of node-sass is not used here, which means I need to update my servers manually every single time the node-sass version changes! And if I want to build separate projects that use slightly different node-sass versions, I can't use the same server.

### Consistent behaviour!
I think `sass-binary-site` and `sass-binary-dir` should behave similarly, just with one providing a website and the other a directory.
Ideally, I'd like to be able to configure more about this process (for instance, not have the directory structure but use the version in the filename), but at the very least behaviour should be consistent.

#### Info for Windows 10 system
- NPM version (`npm -v`): 
`5.6.0`
- Node version (`node -v`): 
`v8.11.2`
- Node Process (`node -p process.versions`):
```
{ http_parser: '2.8.0',
 node: '8.11.2',
 v8: '6.2.414.54',
 uv: '1.19.1',
 zlib: '1.2.11',
 ares: '1.10.1-DEV',
 modules: '57',
 nghttp2: '1.29.0',
 napi: '3',
 openssl: '1.0.2o',
 icu: '60.1',
 unicode: '10.0',
 cldr: '32.0',
 tz: '2017c' }
```
- Node Platform (`node -p process.platform`):
`win32`
- Node architecture (`node -p process.arch`):
`ia32`
- node-sass version (`node -p "require('node-sass').info"`):
`4.9.2`
- npm node-sass versions (`npm ls node-sass`):
```
angularIB@0.0.0 /project-path
└─┬ @angular-devkit/build-angular@0.7.2
  └── node-sass@4.9.2 
```

#### Info for Mac OS Mojave system
- NPM version (`npm -v`): 
`6.4.1`
- Node version (`node -v`): 
`v10.15.0`
- Node Process (`node -p process.versions`):
```
{ http_parser: '2.8.0',
  node: '10.15.0',
  v8: '6.8.275.32-node.45',
  uv: '1.23.2',
  zlib: '1.2.11',
  ares: '1.15.0',
  modules: '64',
  nghttp2: '1.34.0',
  napi: '3',
  openssl: '1.1.0j',
  icu: '62.1',
  unicode: '11.0',
  cldr: '33.1',
  tz: '2018e' }
```
- Node Platform (`node -p process.platform`):
`darwin`
- Node architecture (`node -p process.arch`):
`x64`
- node-sass version (`node -p "require('node-sass').info"`):
`4.9.2`
- npm node-sass versions (`npm ls node-sass`):
```
angularIB@0.0.0 /project-path
└─┬ @angular-devkit/build-angular@0.7.2
  └── node-sass@4.9.2 
```

**Comment by nschonni** [comment]
You're right. I'm hoping to replace our custom binary logic with https://github.com/mapbox/node-pre-gyp for v5, so I'm not sure if this will be addressed in a release

**Comment by cunningt** [comment]
I tried to make this work through sass-binary-site and file:/// URI to no avail, so it'd be nice if sass-binary-dir offered the version disambiguation.

```
[INFO] verbose 43.273 Downloading binary from file://localhost/opt/build-agent/.npm/node-sass/v4.11.0/linux-x64-57_binding.node
[INFO] Cannot download "file://localhost/opt/build-agent/.npm/node-sass/v4.11.0/linux-x64-57_binding.node": 
[INFO] 
[INFO] getaddrinfo ENOTFOUND http http:80
```
